### PR TITLE
Fetch minischedules data in frontend 

### PR DESCRIPTION
**Pull request by skyqrose**
Asana Task: [Mini-schedule | Frontend | Fetch data](https://app.asana.com/0/1148853526253426/1170680075334329)

Screen Shot 2020-04-29 at 14 09 38
Screen Shot 2020-04-29 at 14 09 47

### CODE REVIEW

File: assets/src/models/minischeduleData.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      : pieceFromData(activityData)
+  ),
+})
+
+export const blockFromData = (blockData: BlockData): Block => ({
+  id: blockData.id,
+  pieces: blockData.pieces.map(pieceFromData),
+})
+
+export const breakFromData = (breakData: BreakData): Break => ({
```
Comment (arkadyan) [inline assets/src/models/minischeduleData.ts:70]: Not a big deal, but several of these functions are exported but not used outside of this file.

File: assets/src/components/propertiesPanel/status_run_block_tabs.tsx [hunk 2]
```diff
@@ -49,13 +52,13 @@ const StatusRunBlockTabs = ({ statusContent }: Props) => (
       className="m-tabs__tab-panel"
       selectedClassName="m-tabs__tab-panel--selected"
     >
-      <h2>Run Content</h2>
+      {activeTripId && <MinischeduleRun activeTripId={activeTripId} />}
```
Comment (arkadyan) [inline assets/src/components/propertiesPanel/status_run_block_tabs.tsx:55]: This is likely a question for Ingrid, but if there is no active trip, do we want to display some message, or not show tabs in the first place or something?

File: assets/src/models/minischeduleData.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+const isBreakData = (
+  activityData: BreakData | PieceData
+): activityData is BreakData => activityData.hasOwnProperty("break_type")
+
+export const runFromData = (runData: RunData): Run => ({
+  id: runData.id,
+  activities: runData.activities.map((activityData) =>
+    isBreakData(activityData)
+      ? breakFromData(activityData)
+      : pieceFromData(activityData)
```
Comment (arkadyan) [inline assets/src/models/minischeduleData.ts:61]: It looks like this branch isn't being tested yet?

File: assets/src/components/propertiesPanel/minischedule.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+} from "../../hooks/useMinischedule"
+import { Block, Run } from "../../minischedule"
+import { TripId } from "../../schedule"
+import Loading from "../loading"
+
+export interface Props {
+  activeTripId: TripId
+}
+export const MinischeduleRun = ({ activeTripId }: Props) => {
+  const run: Run | null = useMinischeduleRun(activeTripId)
```
Comment (arkadyan) [inline assets/src/components/propertiesPanel/minischedule.tsx:14]: Waiting to test these two until we have actual content there seems fine by me.

File: assets/src/hooks/useMinischedule.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from "react"
+import { fetchMinischeduleBlock, fetchMinischeduleRun } from "../api"
+import { Block, Run } from "../minischedule"
+import { TripId } from "../schedule"
+
+export const useMinischeduleRun = (tripId: TripId): Run | null => {
+  const [run, setRun] = useState<Run | null>(null)
+  useEffect(() => {
+    fetchMinischeduleRun(tripId).then(setRun)
```
Comment (arkadyan) [inline assets/src/hooks/useMinischedule.ts:9]: These API calls can return `null` as a legitimate response, right? Do we want to set the state to be `undefined` initially to distinguish between loading and empty results?

File: assets/src/hooks/useMinischedule.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  }, [])
```
Comment (arkadyan) [inline assets/src/hooks/useMinischedule.ts:10]: Should this `useEffect` (and below) pass `tripId` as a dependency? If the vehicle's trip changes while this panel is open we'd want to load fresh data, right?

Comment (skyqrose) [reply assets/src/hooks/useMinischedule.ts:10]: Good catch!

File: assets/src/components/propertiesPanel/status_run_block_tabs.tsx (hunk 2 above)
Comment (skyqrose) [reply assets/src/components/propertiesPanel/status_run_block_tabs.tsx:55]: @ingridpierre ?

File: assets/src/components/propertiesPanel/minischedule.tsx (hunk 4 above)
Comment (skyqrose) [reply assets/src/components/propertiesPanel/minischedule.tsx]: I ended up adding tests, since the loading/null/data conditionals are reasonably permanent, even if the content of the snapshot will change soon.

File: assets/src/models/minischeduleData.ts (hunk 3 above)
Comment (skyqrose) [reply assets/src/models/minischeduleData.ts:61]: `pieceFromData` is tested by the the piece in the block test.
